src/timing.c

```diff
@@ -78,12 +83,29 @@ void jl_print_timings(void)
 #endif
 }
 
+
 void jl_init_timing(void)
 {
     t0 = cycleclock();
 
-    _Static_assert(JL_TIMING_EVENT_LAST < sizeof(uint64_t) * CHAR_BIT, "Too many timing events!");
-    _Static_assert((int)JL_TIMING_LAST <= (int)JL_TIMING_EVENT_LAST, "More owners than events!");
+    _Static_assert(JL_TIMING_SUBSYSTEM_LAST < sizeof(uint64_t) * CHAR_BIT, "Too many timing subsystems!");
+
+#ifdef USE_TIMING_COUNTS
+    // Create events list for counts backend
+    arraylist_new(&jl_timing_counts_events, 1);
+
+    jl_timing_counts_event_t *new_event = (jl_timing_counts_event_t *)malloc(sizeof(jl_timing_counts_event_t));
+    arraylist_push(&jl_timing_counts_events, (void *)new_event);
+
+    new_event->name = "ROOT";
+    jl_atomic_store_relaxed(&new_event->total, 0);
+#endif
+
+    // Sort the subsystem names for quick enable/disable lookups
+    qsort(
+        jl_timing_subsystems, JL_TIMING_SUBSYSTEM_LAST,
+        sizeof(const char *), (int(*)(const void*, const void*))strcmp
```

Comment:
There shouldn't be a need to cast strcmp?

Reply:
```
/home/topolarity/repos/julia/src/timing.c:107:31: warning: passing argument 4 of 'qsort' from incompatible pointer type [-Wincompatible-pointer-types]
  107 |         sizeof(const char *), strcmp
      |                               ^~~~~~
      |                               |
      |                               int (*)(const char *, const char *)
In file included from /home/topolarity/repos/julia/src/support/libsupport.h:8,
                 from /home/topolarity/repos/julia/src/julia.h:19,
                 from /home/topolarity/repos/julia/src/timing.c:4:
/usr/include/stdlib.h:839:34: note: expected '__compar_fn_t' {aka 'int (*)(const void *, const void *)'} but argument is of type 'int (*)(const char *, const char *)'
  839 |                    __compar_fn_t __compar) __nonnull ((1, 4));
  ```

Reply:
This was my bad actually.

I had written some optimistic casts planning to test this when I added the Julia layer on top, but there's an extra indirection that I didn't account for here